### PR TITLE
Add test cases for composite keys

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/3954
## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: ballerina/tests/generated_client.bal [hunk 1]
```diff
@@ -6,6 +6,7 @@ const BUILDING = "building";
 const WORKSPACE = "workspace";
 const DEPARTMENT = "department";
 const EMPLOYEE = "employee";
+const ORDERITEM = "orderItem";
```
Comment (daneshk) [inline ballerina/tests/generated_client.bal:9]:
```suggestion
const ORDER_ITEM = "orderitem";
```

Comment (kaneeldias) [reply ballerina/tests/generated_client.bal]: Fixed in [f9dd2cb](https://github.com/ballerina-platform/module-ballerina-persist/pull/105/commits/f9dd2cbe2b6d8409c511f40bf1fd105b7b99858d)

File: ballerina/tests/generated_client.bal [hunk 2]
```diff
@@ -6,6 +6,7 @@ const BUILDING = "building";
 const WORKSPACE = "workspace";
 const DEPARTMENT = "department";
 const EMPLOYEE = "employee";
+const ORDER_ITEM = "orderItem";
```
Comment (daneshk) [inline ballerina/tests/generated_client.bal:9]: Map keys are lowercase in the current impl.

Comment (kaneeldias) [reply ballerina/tests/generated_client.bal]: Updated in [2e6b2be](https://github.com/ballerina-platform/module-ballerina-persist/pull/105/commits/2e6b2be4d839a813171b8e3ef81de106fff8bd3d)

File: ballerina/tests/generated_client.bal [hunk 3]
```diff
@@ -197,6 +210,36 @@ client class RainierClient {
         return 'object;
     };
 
+    isolated resource function get orderItems() returns stream<OrderItem, error?> {
```
Comment (daneshk) [inline ballerina/tests/generated_client.bal:213]: Shall we also change the resource names to lowercase `orderitem`

**Review by daneshk — APPROVED** [review]
Add one more comment. Overall LGTM

File: ballerina/tests/generated_client.bal [hunk 4]
```diff
@@ -197,6 +210,36 @@ client class RainierClient {
...
+               select [inserted.orderId, inserted.itemId];
+    };
+
+    isolated resource function put orderItems/[string orderId]/[string itemId](OrderItemUpdate data) returns OrderItem|error {
+        _ = check self.persistClients.get(ORDER_ITEM).runUpdateQuery({orderId: orderId, itemId: itemId}, data);
+        return self->/orderItems/[orderId]/[itemId].get();
+    };
+
+    isolated resource function delete orderItems/[string orderId]/[string itemId]() returns OrderItem|error {
+        OrderItem 'object = check self->/orderItems/[orderId]/[itemId].get();
```
Comment (niveathika) [inline ballerina/tests/generated_client.bal:238]: Any reason to use `'object` as variable name? IMO result will suffice.

File: ballerina/tests/generated_client.bal [hunk 5]
```diff
@@ -197,6 +210,36 @@ client class RainierClient {
...
+        return (check self.persistClients.get(ORDER_ITEM).runReadByKeyQuery(OrderItem, {orderId: orderId, itemId: itemId})).cloneWithType(OrderItem);
+    };
+
+    isolated resource function post orderItems(OrderItemInsert[] data) returns [string, string][]|error {
+        _ = check self.persistClients.get(ORDER_ITEM).runBatchInsertQuery(data);
+        return from OrderItemInsert inserted in data
+               select [inserted.orderId, inserted.itemId];
+    };
+
+    isolated resource function put orderItems/[string orderId]/[string itemId](OrderItemUpdate data) returns OrderItem|error {
```
Comment (niveathika) [inline ballerina/tests/generated_client.bal:232]:
```suggestion
    isolated resource function put orderItems/[string orderId]/[string itemId](OrderItemUpdate data) returns OrderItem|persist:Error {
```
It should be changed in all resources.

Comment (kaneeldias) [reply ballerina/tests/generated_client.bal]: Fixed in [002ef6f](https://github.com/ballerina-platform/module-ballerina-persist/pull/105/commits/002ef6fc58d62842bfd718458b5063f38333ea5c)

File: ballerina/tests/generated_client.bal (hunk 4 above)
Comment (kaneeldias) [reply ballerina/tests/generated_client.bal]: Fixed in [775f43e](https://github.com/ballerina-platform/module-ballerina-persist/pull/105/commits/775f43eb14f46d103a58633e1e781921f7222a05)

File: ballerina/tests/generated_client.bal (hunk 3 above)
Comment (kaneeldias) [reply ballerina/tests/generated_client.bal]: Fixed in [775f43e](https://github.com/ballerina-platform/module-ballerina-persist/pull/105/commits/775f43eb14f46d103a58633e1e781921f7222a05)

**Review by niveathika — APPROVED** [review]
LGTM. Seems there are changes to generated client. Lets coordinate with @sahanHe to reflect the changes to the clients

**Pull request merged**